### ISSUE TITLE
`vue/no-bare-strings-in-template` allow numbers

### ISSUE THREAD

**Issue by mrleblanc101** [vuejs/eslint-plugin-vue#2695]
**What rule do you want to change?**
`vue/no-bare-strings-in-template`

**Does this change cause the rule to produce more or fewer warnings?**
Fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**
Either add an option to allow numbers:
```diff
{
  "vue/no-bare-strings-in-template": ["error", {
+   "allowNumbers": true,
    "allowlist": [
      "(", ")", ",", ".", "&", "+", "-", "=", "*", "/", "#", "%", "!", "?", ":", "[", "]", "{", "}", "<", ">", "\u00b7", "\u2022", "\u2010", "\u2013", "\u2014", "\u2212", "|"
    ],
    "attributes": {
      "/.+/": ["title", "aria-label", "aria-placeholder", "aria-roledescription", "aria-valuetext"],
      "input": ["placeholder"],
      "img": ["alt"]
    },
    "directives": ["v-text"]
  }]
}
```
or add all number to the default allowList:
```js
{
  "vue/no-bare-strings-in-template": ["error", {
    "allowlist": [
      "(", ")", ",", ".", "&", "+", "-", "=", "*", "/", "#", "%", "!", "?", ":", "[", "]", "{", "}", "<", ">", "\u00b7", "\u2022", "\u2010", "\u2013", "\u2014", "\u2212", "|", 0, 1, 2, 3, 4, 5, 6, 7, 8, 9
    ],
    "attributes": {
      "/.+/": ["title", "aria-label", "aria-placeholder", "aria-roledescription", "aria-valuetext"],
      "input": ["placeholder"],
      "img": ["alt"]
    },
    "directives": ["v-text"]
  }]
}
```

**Please provide some example code that this change will affect:**

```vue
<div>1</div>
```

**What does the rule currently do for this code?**
> Unexpected non-translated string used.

**What will the rule do after it's changed?**
No error, number don't need to be "translated" (unless they are formatted, which is not alwats needed)

**Additional context**

**Comment by mrleblanc101** [comment]
If there was a way to append to the `allowlist` without overwritting it, maybe we wouldn't need this.
Maybe it possible, but I'm not sure how.

```
'vue/no-bare-strings-in-template': [
    'error',
    {
        allowlist: [ ...allowlist, 1, 2, 3, 4, 5, 6, 7, 8, 9 ],
        attributes: {
            '/.+/': [
                'title',
                'description',
                'label',
                'help',
                'tooltip',
                'placeholder',
                'errors',
                'content',
                'confirmButtonLabel',
                'cancelButtonLabel',
            ],
            'UTab': ['name'],
        },
        directives: ['v-text'],
    },
],
```

**Comment by waynzh** [comment]
> allow numbers

[The proposal](https://github.com/vuejs/eslint-plugin-vue/issues/2613#issuecomment-2538160327) to allow regexp patterns in the `allowlist` seems to address this issue.

> extends the default `allowlist` without overwriting it

Hmm, as far as I know, currently there's none.

**Comment by waynzh** [comment]
I'll close this issue for now, since [v10.1.0](https://github.com/vuejs/eslint-plugin-vue/releases/tag/v10.1.0) supports regexp.
